### PR TITLE
Fitting

### PR DESCRIPTION
**Pull request by mehradans92**
- Added ref-error metric.

- Added two notebooks for data fitting (still requires updates).

- Revised the contact_infection_func.

- Added compartment restrainer

Please review the changes to **contact_infection_func** and the fitting in the **NeurIPS** notebook.

### CODE REVIEW

File: maxentep/maxent.py [hunk 1]
```diff
@@ -209,6 +231,18 @@ def __init__(self, restraints, use_cov=False, name='maxent-model', **kwargs):
             self.avg_layer = AvgLayer(self.weight_layer)
         self.lambdas = self.weight_layer.l
         self.prior = prior
+        if ref_traj is not None:
+            self.ref_traj = ref_traj
+            if population_fraction is not None:
```
Comment (whitead) [inline maxentep/maxent.py:236]: Why is  this indented? Shouldn't it not be affected by ref_traj?

File: maxentep/maxent.py [hunk 2]
```diff
@@ -248,16 +289,29 @@ def reweight(samples, unbiased_joint, joint):
 
 class HyperMaxentModel(MaxentModel):
     def __init__(self, restraints, prior_model, simulation, reweight=True,
-                 name='hyper-maxent-model', **kwargs):
+                 name='hyper-maxent-model', ref_traj=None, trajs=None, population_fraction=None, ** kwargs):
```
Comment (whitead) [inline maxentep/maxent.py]: Why not allow population_fraction and ref_traj to be passed to MaxentModel?

File: maxentep/maxent.py [hunk 3]
```diff
@@ -248,16 +289,29 @@ def reweight(samples, unbiased_joint, joint):
...
             restraints=restraints, name=name, **kwargs)
         self.prior_model = prior_model
         self.reweight = reweight
         self.unbiased_joint = prior_model(tf.constant([1.]))
+        self.trajs = trajs
         if hasattr(self.unbiased_joint, 'sample'):
             self.unbiased_joint = [self.unbiased_joint]
         self.simulation = simulation
-
+        if ref_traj is not None:
```
Comment (whitead) [inline maxentep/maxent.py]: Just let parent class, MaxEntModel do this. Do not replicate the code. This will be executed when you call super.__init__

**Review by whitead — CHANGES_REQUESTED** [review]
Looks good, see in code comments. Also, can you rename jupyter notebooks to be more descriptive (Fit_data and NeurIPS).

File: maxentep/maxent.py (hunk 1 above)
Comment (mehradans92) [reply maxentep/maxent.py:236]: No, it should not. If the user inputs ref_traj, the model will display the ref_error metric during the fit. In this case, if no inputs are provided for the population fraction, a uniform population fraction is considered among all the patches for calculating the weighted average trajectory.

**Review by whitead — APPROVED** [review]

**Pull request merged**